apps/language_server/lib/language_server/providers/execute_command/manipulate_pipes.ex

```diff
@@ -0,0 +1,274 @@
+defmodule ElixirLS.LanguageServer.Providers.ExecuteCommand.ManipulatePipes do
+  @moduledoc """
+  This module implements a custom command for converting function calls
+  to pipe operators and pipes to function calls.
+
+  Returns a formatted source fragment.
+  """
+  import ElixirLS.LanguageServer.Protocol
+
+  alias ElixirLS.LanguageServer.{JsonRpc, Server}
+
+  alias __MODULE__.AST
+
+  @behaviour ElixirLS.LanguageServer.Providers.ExecuteCommand
+
+  @impl ElixirLS.LanguageServer.Providers.ExecuteCommand
+  def execute(
+        %{"uri" => uri, "cursor_line" => line, "cursor_column" => col, "operation" => operation},
+        state
+      )
+      when is_integer(line) and is_integer(col) and is_binary(uri) and
+             operation in ["to_pipe", "from_pipe"] do
+    # line and col are assumed to be 1-indexed
```

Comment:
While a bit awkward, for consistency we should assume these are both 0-indexed because that's the standard that LSP uses (as well as the expandMacro command):
> A position inside a document (see Position definition below) is expressed as a zero-based line and character offset. The offsets are based on a UTF-16 string representation.

https://microsoft.github.io/language-server-protocol/specification#textDocuments

Reply:
No problem. As I didn't have clear standard to follow, I assumed 1 indexed because VSCode shows them as 1-indexed in the  bottom toolbar. However, changing this I think is just a matter of subtracting 1 somewhere, with no changes to core functionality

Reply:
@axelson I've implemented both this change and the test assertion order. Hopefully it's ready to go now!

Reply:
Nevermind, I still have to fix the tests which are breaking due to versioning. I'll add the parens to the original AST so we don't have to deal with this minor detail

Reply:
All tests pass now :D

Reply:
Great, thanks!